app/models/pawn.rb

```diff
@@ -1,8 +1,36 @@
 class Pawn < Piece
-  def first_move
-    # can move two spaces forward but only on first move
+  def valid_move?(x, y)
+    is_obstructed?(x, y)
+    one_forward(x, y) || two_forward(x, y) || one_diagonal(x, y)
   end
 
-  def move
+  def white_moving
+    color
+  end
+
+  def one_forward(x, y)
+    if white_moving
+      x == x_position && y == y_position + 1
+    else
+      x == x_position && y == y_position - 1
+    end
+  end
+
+  def two_forward(x, y)
+    if white_moving
+      y_position == 1 && x == x_position && y == y_position + 2
+    else
+      y_position == 6 && x == x_position && y == y_position - 2
+    end
+  end
+
+  def one_diagonal(x, y)
+    unless game.pieces.where(:x_position => x, :y_position => y).nil?
+      if white_moving
+        y == y_position + 1 && (x == x_position + 1 || x == x_position - 1)
+      else
+        y == y_position - 1 && (x == x_position + 1 || x == x_position - 1)
+      end
+    end
   end
 end
```

Comment:
This is a beautiful and simple implementation @nicolaguidi. It appears you cover all valid and failure cases.  My only comment is to suggest you consider using an absolute delta to determine move validity because absolute deltas cover positive and negative vectors without being concerned about starting color or direction of movement.  If in future cases the board layout changes or the board orientation changes for user perspective this may make your code more flexible.  That said, if you don't make any changes I am ready to merge this as-is.  Tests and logic are great!  :+1: